src/components/Modals/DeleteClientModal.jsx

```diff
@@ -26,7 +26,7 @@ import FormSection from '../Form/FormSection';
  */
 
 const DeleteClientModal = ({
-  showDeleteClientModal,
+  showDeleteClientModal = false,
```

Comment:
The Dialog Element used by MUI expects to always receive a value. If it receives `null` or `undefined` it generates a warning. If `showDeleteClientModal` isn't passed in, this would previously have been `undefined`. Now we default it to false.

Reply:
Can you explain this a little more (really just to better my understanding)... the state is instantiated as `false` to begin with. It's never `null` or `undefined`, only `true` or `false` - or am I confidently incorrect?

Reply:
This only happened in the test. The test didn't pass in `showDeleteClientModal={false}` when rendering `DeleteClientModal`. That left `showDeleteClientModal` as `undefined` when passed to the MUI `Dialog` component. Which caused that ugly stack trace in the test output.

Reply:
hmmmm...   
**1)** shouldn't `showDeleteClientModal` be `true` if we want `DeleteClientModal` to render?
**2)** so why wouldn't we do this in the test file instead of the source component file?

Like here on `line 77` and `line 83`:
<img width="843" alt="Screenshot 2023-07-24 at 9 04 23 PM" src="https://github.com/codeforpdx/PASS/assets/71395931/cad3dcf3-b505-4667-8a49-fdc9a512a0bc">



Again, I'm just trying to understand. I'm just not following the logic here yet.

Reply:
If we wanted `showDeleteClientModal` to render, then yes, we would. But the relevant test case is lines 34-49:
```js
it('hides/unmounts when showDeleteClientModal is false', () => {
  const removeUser = vi.fn();
  const setModal = () => {};
  const client = { person: 'john' };
  const { queryByLabelText } = render(
    <UserListContext.Provider value={{ removeUser }}>
      <DeleteClientModal setShowDeleteClientModal={setModal} selectedClientToDelete={client} />
    </UserListContext.Provider>
  );

  const deleteButton = queryByLabelText('Delete Client Button');
  const cancelButton = queryByLabelText('Cancel Button');

  expect(deleteButton).toBeNull();
  expect(cancelButton).toBeNull();
});
```

In that one we want the opposite. We _don't_ want to show the modal. So we pass nothing. The nothing (or `undefined`) gets coerced to `false`, but generates a warning. 

You can see this for yourself. Remove this change, then run test case `hides/unmounts when showDeleteClientModal is false`. You'll see the warning generate. You can then look through the provided stack trace and see what it's complaining about.
---

The changes here aren't major. I just found the warnings annoying.

Reply:
Ok, and we can't just pass it in? `showDeleteClientModal={false}` because then it won't render the component at all?

I guess I'm trying to understand why we cannot do whatever needs to be done in the test file (if this is only an issue for the tests) or why the state that controls whether the modal is rendered or not isn't being passed in, in the first place.

I'm missing something clearly.

I don't disagree that these changes are good or minimal, I don't even disagree that they should be done... I'm just trying to understand the approach.


Also, `App.test.jsx` fails - but I haven't looked into it yet.